compiler/rustc_resolve/src/lib.rs

```diff
@@ -2639,7 +2639,7 @@ impl<'a> Resolver<'a> {
                                         self.report_error(
                                             span,
                                             ResolutionError::ParamInNonTrivialAnonConst(
```

Comment:
you are dealing with a type parameter here, so we want to use `None` here.

Reply:
As discussed out-of-band, we needed the name of the symbol here, so we used an additional bool, and passed it in.